<launch>
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <rosparam param="model" subst_value="True">$(arg model)</rosparam>

  <arg name="automate_movement" default="False"/>
  <rosparam param="automate_movement" subst_value="True">$(arg automate_movement)</rosparam> 

  <arg name="start_x_pos" default="1.0"/>
  <arg name="start_y_pos" default="1.0"/>
  <arg name="start_yaw" default="$(eval 3.14159/2)"/> 
  <rosparam param="start_pos" subst_value="True">[$(arg start_x_pos), $(arg start_y_pos), $(arg start_yaw)]</rosparam>

  <arg name="robot_radius" default="0.105"/>
  <rosparam param="robot_radius" subst_value="True">$(arg robot_radius)</rosparam>

  <arg name="wheel_radius" default="0.033"/>
  <rosparam param="wheel_radius" subst_value="True">$(arg wheel_radius)</rosparam>

  <arg name="robot_width" default="0.038"/>
  <rosparam param="robot_width" subst_value="True">$(arg robot_width)</rosparam>

  <arg name="diff_drive_time" default="0.1"/>
  <rosparam param="diff_drive_time" subst_value="True">$(arg diff_drive_time)</rosparam>  

  <arg name="origin_offset_x" default="5.0"/>
  <arg name="origin_offset_y" default="5.0"/>
  <rosparam param="origin_offset" subst_value="True">[$(arg origin_offset_x), $(arg origin_offset_y)]</rosparam>

  <arg name="goal_x_pos" default="5.0"/>
  <arg name="goal_y_pos" default="8.0"/>
  <arg name="goal_yaw" default="0.0"/>
  <rosparam param="goal_pos" subst_value="True">[$(arg goal_x_pos), $(arg goal_y_pos), $(arg goal_yaw)]</rosparam>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find path_planning)/worlds/course.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <arg name="abs_start_x_pos" default="$(eval start_x_pos-origin_offset_x)"/>
  <arg name="abs_start_y_pos" default="$(eval start_y_pos-origin_offset_y)"/>

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_$(arg model) -x $(arg abs_start_x_pos) -y $(arg abs_start_y_pos) -z 3.0 -Y $(arg start_yaw) -param robot_description" />

  <node pkg="path_planning" type="path_planning_node.py" name="path_planning_node" output="screen"/>
</launch>
